src/vos/vos_aggregate.c

```diff
@@ -1906,7 +1908,11 @@ join_merge_window(daos_handle_t ih, struct agg_merge_window *mw,
 	/* Lookup physical entry, enqueue if it doesn't exist */
 	phy_ent = lookup_phy_ent(mw, &phy_ext, entry);
 	if (phy_ent == NULL) {
-		D_ASSERT(phy_ext.ex_lo == lgc_ext.ex_lo);
+		if (phy_ext.ex_lo != lgc_ext.ex_lo) {
+			/** Tail of a physical entry that was already removed */
+			D_ASSERT(!visible);
```

Comment:
I think the 20 and 4 case hits it in the new test.  It happens when the physical record is in the merge window but the middle is covered by a removal record that triggers a flush. Maybe I need to look at that flush trigger and see if it can be improved somewhat.

I could assert on partial but isn't that implied by the lo not being equal?